File: src/models/resource-content.ts
import type { ResourceContentCategory } from "./resource-content-category";
import type { AssignmentTheme } from "./assignment-theme";
import type { Attachment } from "./attachment";

export type ResourceContent = Readonly<{
  id: string;

  /**
   * @remark Optional because teachers can just write nothing here and only give a description.
   */
  title?: string;
  /**
   * An HTML string to preserve all the formatting done in the UI.
   * @remark Optional because teachers can just write the title with no description.
   */
  description?: string;

  category: ResourceContentCategory;
  files: Attachment[];
  /** Themes associated with the lesson. */
  themes: AssignmentTheme[];

  /** `-1` when not defined. */
  educativeValue: number;
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/ResourceContent.kt
package ink.literate.pawnote.models

data class ResourceContent(
    val id: String,

    /** Optional because teachers can just write nothing here and only give a description. */
    val title: String? = null,
    /**
     * An HTML string to preserve all the formatting done in the UI.
     *
     * Optional because teachers can just write the title with no description.
     */
    val description: String? = null,
    val category: ResourceContentCategory,
    val files: List<Attachment>,
    /** Themes associated with the lesson. */
    val themes: List<AssignmentTheme>,

    /** `-1` when not defined. */
    val educativeValue: Int
)
